Sources/Client/Utils/Atomic.swift

```diff
@@ -56,19 +54,17 @@ public final class Atomic<T> {
     
     /// Update the value safely.
     /// - Parameter changes: a block with changes. It should return a new value.
-    public func update(_ changes: @escaping (T) -> T) {
-        queue.async(flags: .barrier) {
-            guard let oldValue = self.value else {
-                return
+    public func update(_ changes: @escaping (T) -> T?) {
+        queue.async(flags: .barrier) { [weak self] in
+            if let self = self, let oldValue = self.value {
+                self.updateValue(oldValue: oldValue, newValue: changes(oldValue))
             }
-            
-            let newValue = changes(oldValue)
-            self.value = newValue
-            self.valueChanged(newValue, oldValue)
         }
     }
     
-    private func valueChanged(_ newValue: T?, _ oldValue: T?) {
+    private func updateValue(oldValue: T?, newValue: T?) {
```

Comment:
What I mean:
Instead of passing `oldValue` as parameter, it can internally handle it.
Currently:
```swift
updateValue(oldValue: self.value, newValue: newValue)
func updateValue(oldValue: T?, newValue: T?) {
    value = newValue
    ....
    callbackQueue.async { [weak self] in self?.didSet?(newValue, oldValue) }
    ....
}
```
I suggest:
```swift
updateValue(newValue)
func updateValue(_ newValue: T?) {
    let oldValue: T? = self.value
    self.value = newValue
    ...
    callbackQueue.async { [weak self] in self?.didSet?(newValue, oldValue) }
    ...
}
```
because this can be handled inside the function, the function has access to this value anyway.